### PR TITLE
fix route creation on openshift

### PR DESCRIPTION
**Pull request by jkhelil**

### CODE REVIEW

File: pkg/configuration/base/reconcile.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ func (r *JenkinsReconcilerBaseConfiguration) ensureResourcesRequiredForJenkinsDe
 	}
 	r.logger.V(log.VDebug).Info("Jenkins slave Service is present")
 
-	if resources.IsImageRegistryAvailable(r.Client) {
+	if resources.ISRouteAPIAvailable {
```
Comment (akram) [inline pkg/configuration/base/reconcile.go]: rename to `IsRouteAPIAvailable` with uppercase `I` and lowercase `s`

File: pkg/configuration/base/resources/route.go [hunk 2]
```diff
@@ -3,14 +3,17 @@ package resources
...
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
 )
 
 // RouteKind the kind name for route
 const RouteKind = "Route"
 
 var (
-	IsRouteAPIAvailable = false
+	ISRouteAPIAvailable = false
```
Comment (akram) [inline pkg/configuration/base/resources/route.go]: rename to `IsRouteAPIAvailable` with uppercase `I` and lowercase `s`

File: pkg/configuration/base/resources/route.go [hunk 3]
```diff
@@ -29,23 +32,23 @@ func UpdateRoute(actual routev1.Route, jenkins *v1alpha2.Jenkins) routev1.Route
 }
 
 // IsRouteAPIAvailable tells if the Route API is installed and discoverable
-/*func IsRouteAPIAvailable(client client.Client) bool {
-	if routeAPIChecked {
-		return isRouteAPIAvailable
+func IsRouteAPIAvailable(clientSet *kubernetes.Clientset) bool {
+	if RouteAPIChecked {
+		return ISRouteAPIAvailable
```
Comment (akram) [inline pkg/configuration/base/resources/route.go]: rename to `IsRouteAPIAvailable` with uppercase `I` and lowercase `s`

File: pkg/configuration/base/resources/route.go [hunk 4]
```diff
@@ -29,23 +32,23 @@ func UpdateRoute(actual routev1.Route, jenkins *v1alpha2.Jenkins) routev1.Route
...
-		routeAPIChecked = true
-		isRouteAPIAvailable = false
+		RouteAPIChecked = true
+		ISRouteAPIAvailable = false
 	} else {
 		// API Exists
-		routeAPIChecked = true
-		isRouteAPIAvailable = true
+		RouteAPIChecked = true
+		ISRouteAPIAvailable = true
```
Comment (akram) [inline pkg/configuration/base/resources/route.go]: rename to `IsRouteAPIAvailable` with uppercase `I` and lowercase `s`

**Review by akram — CHANGES_REQUESTED** [review]

File: pkg/configuration/base/reconcile.go (hunk 1 above)
Comment (jkhelil) [reply pkg/configuration/base/reconcile.go]: the method is called Is.... thats why I uppercased S

File: pkg/configuration/base/resources/route.go (hunk 2 above)
Comment (jkhelil) [reply pkg/configuration/base/resources/route.go]: same

File: pkg/configuration/base/resources/route.go (hunk 3 above)
Comment (jkhelil) [reply pkg/configuration/base/resources/route.go]: same

File: pkg/configuration/base/resources/route.go (hunk 4 above)
Comment (jkhelil) [reply pkg/configuration/base/resources/route.go]: same

**Pull request merged**